.github/workflows/run-tests.yml

```diff
@@ -0,0 +1,47 @@
+on:
```

Comment:
I should note that I haven't been able to test this workflow. So it might require some tweaking after it's been merged into `main`.